hooks/general_actions.py

```diff
@@ -227,6 +230,25 @@ def execute_action(self, name, params, sg_data):
         elif name == "publish_clipboard":
             self._copy_to_clipboard(sg_data["path"]["local_path"])
 
+        return result
+
+    def execute_entity_doubleclicked_action(self, sg_data):
+        """
+        This action is triggered when an entity is double-clicked.
+        Perform any specific actions and return a dictionary
+        represetnting the entity that the panel will navigate to.
+
+        This base hook method simply returns immediately the entity
+        data of item that was clicked. Override this method to
+        perform any custom handling.
+
+        :param sg_data: Dictionary containing data for the entity that
+                        was double-clicked.
+        :return: A dictionary {"type": 'entity_type', "id": 'entity_id'}
+        """
+
+        return {"type": sg_data["type"], "id": sg_data["id"]}
```

Comment:
You're right, there is no need to return this value since the caller has it already. I introduced this hook so that our VRED integration can do some other actions on double click action, but there is no need to return the entity that is being passed in -- I'll fix this method to have no return value

Reply:
I was also a little confused why I had this returning as well after looking at it again. So the reason I have the entity dict returning is basically to signal to the caller (the dialog) to proceed or cancel the double click action.

In our VRED integration, there are cases where we want to abort the double-click to navigate to the entity in the panel.

I think it would be better to then just return True or False? And the caller has the entity data to navigate, so no need to return that data back.

Reply:
I think True/False for whether to proceed with the double-click action makes sense to me. 👍